docs/ldml/tr35-numbers.md

```diff
@@ -386,36 +386,74 @@ A pattern `type` attribute is used for _compact number formats_, such as the fol
-        <pattern type="1000000" count="other">0 M ¤</pattern>
+        <pattern type="1000000" count="other">0 M ¤</pattern>      …
+    </decimalFormat>
+</decimalFormatLength>
+…
+<currencyFormatLength type="short"> // for illustration below, these are from a different locale
+    <currencyFormat type="standard">
+        <pattern type="1000" count="one">¤0K</pattern>
+        <pattern type="1000" count="one" alt="alphaNextToNumber">¤ 0K</pattern>
+        <pattern type="1000" count="other">¤0K</pattern>
+        <pattern type="1000" count="other" alt="alphaNextToNumber">¤ 0K</pattern>
+        <pattern type="10000" count="one">¤00K</pattern>
+        <pattern type="10000" count="one" alt="alphaNextToNumber">¤ 00K</pattern>
+        <pattern type="10000" count="other">¤00K</pattern>
+        <pattern type="10000" count="other" alt="alphaNextToNumber">¤ 00K</pattern>
+        <pattern type="100000" count="one">¤000K</pattern>
+        <pattern type="100000" count="one" alt="alphaNextToNumber">¤ 000K</pattern>
+        <pattern type="100000" count="other">¤000K</pattern>
+        <pattern type="100000" count="other" alt="alphaNextToNumber">¤ 000K</pattern>
+        <pattern type="1000000" count="one">¤0M</pattern>
+        <pattern type="1000000" count="one" alt="alphaNextToNumber">¤ 0M</pattern>
         …
     </currencyFormat>
 </currencyFormatLength>
 ```
 
 Formats can be supplied for numbers (as above) or for currencies or other units. They can also be used with ranges of numbers, resulting in formatting strings like “$10K” or “$3–7M”.
 
-To format a number N, the greatest type less than or equal to N is used, with the appropriate plural category. N is divided by the type, after removing the number of zeros in the pattern, less 1. APIs supporting this format should provide control over the number of significant or fraction digits.
+To format a number N, use the following steps. The examples use N = 123456, the currency = CAD, and the currency symbol string = "$CA"
+1. Let P be the pattern element with greatest type less than or equal to N, and any count value.
+    * P = `<pattern type="100000" count="**one**">¤000K</pattern>`
+2. Let V be the pattern element value.
+    * V = "¤000K" 
+3. If the element value of P is "0", then use the corresponding non-compact number formatting instead, and skip the rest of these steps — but adjust the precision as described below.
+    * For example, instead of `currencyFormat` `<pattern type="10000" count="one">¤00K</pattern>`, use `<pattern>¤#,##0.00</pattern>`.
+5. If P is a currency format, look at the currency symbol string, and the position of the currency symbol ¤ in the pattern element value.
+If ¤ is immediately to the left of a 0 and the currency string ends with a _letter grapheme cluster_ (eg, $CA),
+or to the right and the currency starts with a _letter grapheme cluster_ (eg, CA$),
+then switch to the `alt=alphaNextToNumber` pattern.
```

Comment:
> ...then switch to the `alt=alphaNextToNumber` pattern.

If there is one (or is this required?).

Reply:
(That isn't actually necessary, since an `alt` element inherits sideways from the non `alt` element, but just for clarity.)